crates/query-engine/metadata/src/metadata/mod.rs

```diff
@@ -15,6 +15,8 @@ use serde::{Deserialize, Serialize};
 #[derive(Clone, Debug, Default, Serialize, Deserialize, JsonSchema)]
 #[serde(rename_all = "camelCase")]
 pub struct Metadata {
+    #[serde(default)]
```

Comment:
I don't think `scalar_types` belong in `Metadata` because the set of scalar types is not itself a root observable/introspected entity but instead derived from the column types and comparison operators etc.

I don't think having the set of scalar types be part of the metadata enables any new behaviors for a user, except maybe the ability to provoke the ndc to expose an inconsistent schema (I'm not sure).

IMO it'd be more appropriate to have the scalar types be part of `RuntimeConfiguration` instead.

Reply:
OK, let's revert this for now and bring it back once `RuntimeConfiguration` is stored rather than re-computed per request.